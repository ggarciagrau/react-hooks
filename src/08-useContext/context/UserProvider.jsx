import { useState } from "react";
import { UserContext } from "./UserContext";

export const UserProvider = ({ children }) => {

  const [user, setUser] = useState({
    id: 0,
    name: "",
    email: ""
  });

  return (
    <UserContext.Provider
      value={{
        user, setUser
      }}
    >
      {children}
    </UserContext.Provider>
  );
};
