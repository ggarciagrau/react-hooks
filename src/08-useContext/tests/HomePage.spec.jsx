import { render } from "@testing-library/react";
import { HomePage } from "../HomePage";
import { UserContext } from "../context/UserContext";

describe("Tests for HomePage", () => {
  it("Should render component without user", () => {
    const rendered = render(
      <UserContext.Provider value={{ user: {} }}>
        <HomePage />
      </UserContext.Provider>
    );

    expect(rendered.container).toMatchSnapshot();
  });

  it("Should render component without user", () => {
    const rendered = render(
      <UserContext.Provider value={{ user: { name: "Roberto" } }}>
        <HomePage />
      </UserContext.Provider>
    );

    
    expect(rendered.container).toMatchSnapshot();
  });
});
