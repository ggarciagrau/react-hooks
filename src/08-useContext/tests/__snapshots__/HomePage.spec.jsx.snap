// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for HomePage Should render component without user 1`] = `
<div>
  <h1>
    HomePage 
    <small />
  </h1>
  <hr />
</div>
`;

exports[`Tests for HomePage Should render component without user 2`] = `
<div>
  <h1>
    HomePage 
    <small>
      Roberto
    </small>
  </h1>
  <hr />
</div>
`;
